src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java

```diff
@@ -49,7 +49,8 @@ public void validateCliDocSections() throws Exception {
         final Map<String, String> cmdDesc = new HashMap<>();
 
         final NodeList sections = getSectionsFromXdoc("src/xdocs/cmdline.xml.vm");
-        final Set<String> cmdOptions = getListById(sections.item(2), "CLI_Options");
+        final Set<String> cmdOptions =
+                getListById(sections.item(2), "Command_line_usage_Command_line_options");
```

Comment:
What is 2 here?

Reply:
@nrmancuso , 2 is the section index of the XML (considering 0-based), I believe.

0. Content
1. Description 
2. **Command Line Usage**

Reply:
@nrmancuso , could you please assign this PR to yourself.

Reply:
> @nrmancuso , could you please assign this PR to yourself.

No need to assign reviewers until CI is green. CI is the first reviewer :)

Reply:
Got it @nrmancuso , please see https://github.com/checkstyle/checkstyle/pull/14799#discussion_r1571327442.

I seem to be falling short on what I'm missing while debug 😕 
Theoretically, the siblings are read only after firstChild is mapped and so on, but not sure why this is being skipped and continued to next node.

Reply:
This PR changes xml a lot, but test that does selection from this xml is not changed. `getSectionsFromXdoc(` should be changed a bit.

Reply:
It use to be:
```
<ul id="CLI_Options">
        <li>
          <code>-c
```
Becomes a way more complicated and different:
```
....
<tbody>
              <tr>
                <td>
                  <a name="c"/>
                  <a href="#c">
                    <strong>-c</strong> configurationFile
                  </a>
```

So getListById should be rewritten.
@MANISH-K-07 , do need help to write a code?

Reply:
Will look into it at the earliest @romani